src/utils/decorators/input-validation/input-validation.js

```diff
@@ -627,31 +692,34 @@ const InputValidation = ComposedComponent => class Component extends ComposedCom
 
     // position icon relative to width of label
     let iconStyle = {};
-
     if (this.props.labelWidth) {
       iconStyle = { [`${this.props.align}`]: `${100 - this.props.labelWidth}%` };
     }
 
     if (this.state.messageLocked) { messageClasses += ' common-input__message--locked'; }
     if (this.flipped) { messageClasses += ' common-input__message--flipped'; }
 
-    return [
-      <Icon
-        key='0'
-        ref={ (validationIcon) => { this.validationIcon = validationIcon; } }
-        type={ type }
-        className={ iconClasses }
-        style={ iconStyle }
-      />,
-      <div key='1' className='common-input__message-wrapper'>
+    /**
+     * TODO Using ReactPortal rather than Portal due to refs
+     * coming back as undefined. Needs investigation.
+     */
+    return (
+      <ReactPortal isOpened>
         <div
-          ref={ (validationMessage) => { this.validationMessage = validationMessage; } }
-          className={ messageClasses }
+          className='common-input__validation'
+          ref={(node) => { this._portal = node; }}
         >
-          { this.state.errorMessage || this.state.warningMessage || this.state.infoMessage }
+          <Icon
+            key='0'
+            ref={ (validationIcon) => { this.validationIcon = validationIcon; } }
+            type={ type }
+            className={ iconClasses }
+            style={ iconStyle }
+          />
```

Comment:
does the icon need to be fed through the portal? can't it just render in position? (maybe this is why you need the input ref?)